pyg_lib/csrc/ops/cuda/matmul_kernel.cu

```diff
@@ -103,30 +103,28 @@ int shared_memory_for_kernel() {
 
 // Returns the bytes of shared memory available per SM on the GPU, or -1 on
 // error.
-int shared_memory_per_sm() {
+cudaDeviceProp get_dev_prop() {
   cudaDeviceProp properties;
   int device_idx;
   cudaError_t result = cudaGetDevice(&device_idx);
   if (result != cudaSuccess) {
-    return -1;
+    throw std::runtime_error(cudaGetErrorString(result));
   }
 
   result = cudaGetDeviceProperties(&properties, device_idx);
   if (result != cudaSuccess) {
-    return -1;
+    throw std::runtime_error(cudaGetErrorString(result));
   }
-
-  return properties.sharedMemPerMultiprocessor;
+  return properties;
+}
+cudaDeviceProp props;
+if (torch::cuda::is_available()) {
+  props = get_dev_prop();
```

Comment:
Doesn't this get invoked directly on import? Wouldn't it be more reasonable to cache it instead, e.g., via `static`?

Reply:
yeah i noticed thats why the CI was failing. will fix